examples/util/RSocket.h

```diff
@@ -0,0 +1,36 @@
+// Copyright 2004-present Facebook. All Rights Reserved.
+
+#pragma once
+
+#include <folly/ExceptionWrapper.h>
+#include <folly/futures/Future.h>
+#include "examples/util/TcpConnectionFactory.h"
+#include "src/Payload.h"
+#include "src/ReactiveStreamsCompat.h"
+#include "src/StandardReactiveSocket.h"
+
+using namespace ::reactivesocket;
+
+/**
+ * Simplified API for client/server
+ */
+namespace rsocket {
+
+class RSocket {
+ public:
+  static std::unique_ptr<RSocket> createClientFactory(
+      std::unique_ptr<ConnectionFactory> connection);
+  RSocket(std::unique_ptr<ConnectionFactory> connection);
+  ~RSocket();
+  /*
+   * Connect asynchronously and return a Future
```

Comment:
Making connect in itself async has some downsides like not allowing fast start as I can't issue requests before/after the connect call happens (and before it returns).

This is also something we currently depend on for clients, where we'll kick of the connect() call on the first subscription from user code but still allow all of of them to transparently succeed.